lib/components/viewers/vehicle-position-retriever.ts

```diff
@@ -0,0 +1,72 @@
+import { connect } from 'react-redux'
+import { useCallback, useEffect, useState } from 'react'
+
+import * as apiActions from '../../actions/api'
+
+interface Props {
+  getVehiclePositionsForRoute: (id: string) => void
+  refreshSeconds: number
+  routeId?: string
+}
+
+/**
+ * Non-visual component that retrieves vehicle positions for the given route.
+ */
+const VehiclePositionRetriever = ({
+  getVehiclePositionsForRoute,
+  refreshSeconds,
+  routeId
+}: Props) => {
+  const [refreshTimer, setRefreshTimer] = useState<NodeJS.Timeout | null>(null)
+
+  const refreshVehiclePositions = useCallback(() => {
+    if (routeId) {
+      getVehiclePositionsForRoute(routeId)
+    }
+  }, [routeId, getVehiclePositionsForRoute])
+
+  useEffect(() => {
+    // Fetch vehicle positions when initially mounting component and a route id is available.
+    if (routeId) {
+      refreshVehiclePositions()
+
+      if (!refreshTimer) {
+        console.log('Set up timer for route ' + routeId || '')
```

Comment:
Can we change the verb here to match "unmounting" below?
```suggestion
        console.log('Setting up timer for route ' + routeId || '')
```

Reply:
Actually I forgot to remove these log statements!

Reply:
Fixed in https://github.com/opentripplanner/otp-react-redux/pull/848/commits/16c0707acde81511e8ef7aa4da4e4919869d76a0.